docs/csharp/tutorials/generate-consume-asynchronous-stream.md

```diff
@@ -0,0 +1,120 @@
+
+## Convert to async streams
+
+Next, convert the `runPagedQueryAsync` to generate an async stream. First, change the signature of `runPagedQueryAsync` to return an `IAsyncEnumerable<JToken>`, and remove the cancellation token and progress objects from the parameter list as shown in the following code:
+
+[!code-csharp[FinishedSignature](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#UpdateSignature)]
+
+The starter code processes each page as the page is retrieved, as shown in the following code:
+
+[!code-csharp[StarterPaging](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#ProcessPage)]
+
+Replace those three lines with the following code:
+
+[!code-csharp[FinishedPaging](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#YieldReturnPage)]
+
+You can also remove the declaration of `finalResults` earlier in this method, and the return statement that follows the loop you modified.
+
+You've finished the changes to generate an async stream. The finished method should resemble the code below:
+
+[!code-csharp[FinishedGenerate](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#GenerateAsyncStream)]
+
+Next, you change the code that consumes the collection to consume the async stream. Find the following code in `Main` that processes the collection of issues:
+
+[!code-csharp[EnumerateOldStyle](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#EnumerateOldStyle)]
+
+Replace all that code with the following `await foreach` loop:
+
+[!code-csharp[FinishedEnumerateAsyncStream](../../../samples/csharp/tutorials/AsyncStreams/finished/IssuePRreport/IssuePRreport/Program.cs#EnumerateAsyncStream)]
+
+Run the application again. Contrast its behavior with the behavior of the starter application. The first page of results is enumerated as soon as they are available. There's an observable pause as each new page is requested and retrieved, then the next page's results are quickly enumerated. The `try` / `catch` block is not needed to handle cancellation: the caller could stop enumerating the collection. Progress is clearly reported because the async stream generates results as each page is downloaded.
+
+You can see improvements in memory use by examining the code. You no longer need to allocate a collection to store all the results before they are enumerated. The caller can determine how to consume the results and if a storage collection is needed.
+
+You can get the code for the finished tutorial from our [samples](https://github.com/dotnet/samples) repository in the [csharp/tutorials/AsyncStreams](https://github.com/dotnet/samples/tree/master/csharp/tutorials/AsyncStreams/finished) folder.
```

Comment:
```suggestion
You can get the code for the finished tutorial from the [dotnet/samples](https://github.com/dotnet/samples) repository in the [csharp/tutorials/AsyncStreams](https://github.com/dotnet/samples/tree/master/csharp/tutorials/AsyncStreams/finished) folder.
```

Reply:
I think that this might be better placed immediately after teh "Convert to async streams" header.